src/app.js

```diff
@@ -81,8 +81,16 @@ export function app(app) {
             }).data
           )
 
-          if (result != null && result.then == null) {
-            repaint((state = merge(state, emit("update", result))))
+          function update(data) {
+            if (data != null) {
+              repaint((state = merge(state, emit("update", data))))
+            }
+          }
+
+          if (result != null && typeof result.then == "function") {
+            result.then(update)
+          } else {
+            update(result)
           }
```

Comment:
So, we need to check if `resul.then` is a function in order to call it.

If the end user return a function let execute it, and the result not null check will do the rest. If this is not a function lets do a regular result patch.

Without that, using `then` in result will break the app with an hyperapp internal error.